common/src/services/auth.service.ts

```diff
@@ -280,7 +280,7 @@ export class AuthService implements AuthServiceAbstraction {
 
         let emailPassword: string[] = [];
         let codeCodeVerifier: string[] = [];
-        let clientIdClientSecret: string[] = [];
+        let clientIdClientSecret: [string, string] = [null, null];
```

Comment:
Does this break the `TokenRequest` [constructor](https://github.com/bitwarden/jslib/blob/master/common/src/models/request/tokenRequest.ts#L18)? It seems like there are a few more down and upstream impacts/dependencies here on the clientId/secret structure, unless this type satisfies the token request server needs (an array that could be populated as `[null, null]` vs. `[]` or just `null`.

Reply:
If you mean in terms of typescript, no. It's a more permissive to less permissive type.

If you mean in terms of communication with the server project. It shouldn't. All that I'm doing is removing `[]` from a possibility in your list, which isn't used as the code exists now. I just tested login out with `clientIdClientSecret = [null, null]` and everything worked as expected.

Reply:
I'm fine either editing `tokenRequest` to expect a `[string, string]` or changing my code to take a `string[]` if you're worried about it.

Reply:
Nope, that's all good, thanks for the clarification and test/check!